# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/android/nsWindow.cpp
# Commit: 384b025f56d3
# Full Hash: 384b025f56d3f87e045ecede6b1a16b71c0b5d4f
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 21 - Separate nsIDragService and nsIDragSession implementations r=gstoll,geckoview-reviewers,rkraesig,win-reviewers,m_kato
#   
#   Split the class inheritance trees for nsIDragService and nsIDragSession.
#   Remember that, before the start of this patch series, the inheritance
#   diagram was:
# ==============================================================================

diff -r beeda5ef3712 -r 384b025f56d3 widget/android/nsWindow.cpp
--- a/widget/android/nsWindow.cpp	Thu Jul 04 07:48:11 2024 +0000
+++ b/widget/android/nsWindow.cpp	Thu Jul 04 07:48:11 2024 +0000
@@ -2677,15 +2677,16 @@
                            jni::Object::Param aDropData) {
   MOZ_ASSERT(NS_IsMainThread());
 
+  LayoutDeviceIntPoint point =
+      LayoutDeviceIntPoint(int32_t(floorf(aX)), int32_t(floorf(aY)));
+
   RefPtr<nsDragService> dragService = nsDragService::GetInstance();
   if (!dragService) {
     return;
   }
 
-  LayoutDeviceIntPoint point =
-      LayoutDeviceIntPoint(int32_t(floorf(aX)), int32_t(floorf(aY)));
-
-  nsCOMPtr<nsIDragSession> dragSession = dragService->GetCurrentSession(this);
+  RefPtr<nsDragSession> dragSession =
+      static_cast<nsDragSession*>(dragService->GetCurrentSession(this));
   if (dragSession && aAction == java::sdk::DragEvent::ACTION_DRAG_STARTED) {
     dragSession->SetDragEndPoint(point.x, point.y);
     return;
@@ -2702,10 +2703,13 @@
     nsIWidget* widget = this;
     dragService->StartDragSession(widget);
     // For compatibility, we have to set temporary data.
+    dragSession =
+      static_cast<nsDragSession*>(dragService->GetCurrentSession(this));
+    MOZ_ASSERT(dragSession);
+
     auto dropData =
         mozilla::java::GeckoDragAndDrop::DropData::Ref::From(aDropData);
-    nsDragService::SetDropData(dropData);
-    dragSession = dragService->GetCurrentSession(this);
+    dragSession->SetDropData(dropData);
   }
 
   if (dragSession) {
@@ -2727,7 +2731,7 @@
         }
         auto dropData =
             mozilla::java::GeckoDragAndDrop::DropData::Ref::From(aDropData);
-        nsDragService::SetDropData(dropData);
+        dragSession->SetDropData(dropData);
         dragSession->SetDragEndPoint(point.x, point.y);
         break;
       }